sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala

```diff
@@ -114,6 +114,73 @@ case class IsVariantNull(child: Expression) extends UnaryExpression
     copy(child = newChild)
 }
 
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(expr) - Convert a nested input (array/map/struct) into a variant where maps and structs are converted to variant objects which are unordered unlike SQL structs. Input maps can only have string keys.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(named_struct('a', 1, 'b', 2));
+       {"a":1,"b":2}
+      > SELECT _FUNC_(array(1, 2, 3));
+       [1,2,3]
+      > SELECT _FUNC_(array(named_struct('a', 1)));
+       [{"a":1}]
+      > SELECT _FUNC_(array(map("a", 2)));
+       [{"a":2}]
+  """,
+  since = "4.0.0",
+  group = "variant_funcs")
+// scalastyle:on line.size.limit
+case class ToVariantObject(child: Expression)
```

Comment:
Do we have tests that have `null`s in the inputs? I couldn't easily find them in the test files.

`null`s could be anywhere, like as the top-level parameter, or nested in a struct, array, or map.

Reply:
Yes, we do have null testing with a [top-level null](https://github.com/apache/spark/blob/80930845c9f5ddb71e647f648770cc4dfb339bbe/sql/core/src/test/scala/org/apache/spark/sql/VariantEndToEndSuite.scala#L164) and [nested nulls in every nested structure](https://github.com/apache/spark/blob/80930845c9f5ddb71e647f648770cc4dfb339bbe/sql/core/src/test/scala/org/apache/spark/sql/VariantEndToEndSuite.scala#L318).